lib/base.js

```diff
@@ -91,6 +91,8 @@ class BaseObject extends CallableObject {
       options = {};
     } else if (typeof options === 'string') {
       options = { name: options };
+    } else if (typeof options === 'undefined') {
```

Comment:
How are you calling delete?  Are you also leaving `cb` undefined?  If so, won't `this.api.delete` fail when it invokes the cb?

Reply:
Hi! I've been using this lib since version 3 and only update version, but not a code. 
Here is example of using
```javascript
const api = new Api.Api(k8sConfig); //get config from file
this.api = api.group(apiVersion);  //use batch/v1 for jobs
...
await this.api.ns(defaultNS).jobs(jobName).delete();      
```

Reply:
passing empty object into delete method works well, but it isn't documented